testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -0,0 +1,511 @@
+    repl = replicator.create(repl_config)
+    repl_change_listener = replicator.addReplicatorEventChangeListener(repl)
+
+    # 2. Starting Replication and waiting for it finish
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # 3 changing channel of some docs, so that we get access revoked event
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)["rows"]
+    docs_to_modify = random.sample(sg_docs, (num_of_docs / 10) * 2)
+    for sg_doc in docs_to_modify:
+        sg_client.update_doc(url=sg_url, db=sg_db, doc_id=sg_doc["id"],
+                             number_updates=1, auth=auth_session,
+                             channels=[])
+
+    # 4 Replicating access revoked
+    repl_access_revoke_config = replicator.configure(source_db=cbl_db,
+                                                     target_url=sg_blip_url,
+                                                     continuous=False,
+                                                     headers=session_header,
+                                                     replication_type="pull")
+    repl_access_revoke = replicator.create(repl_access_revoke_config)
+    repl_access_revoke_change_listener = replicator.addReplicatorEventChangeListener(repl_access_revoke)
+    replicator.start(repl_access_revoke)
+    replicator.wait_until_replicator_idle(repl_access_revoke)
+
+    # 5. Verifying the access revoke in event captures
+    doc_revoke_access_event_changes = replicator.getReplicatorEventChanges(repl_access_revoke_change_listener)
+    replicator.removeReplicatorEventListener(repl_access_revoke, repl_access_revoke_change_listener)
+    replicator.stop(repl_access_revoke)
+    event_dict = get_event_changes(doc_revoke_access_event_changes)
+    replicator.removeReplicatorEventListener(repl_access_revoke, repl_access_revoke_change_listener)
+    replicator.stop(repl_access_revoke)
+    assert len(event_dict) != 0, "Replication listener didn't caught events. Check app logs for detailed info"
+    for doc_id in event_dict:
+        assert event_dict[doc_id]["flags"] == "2" or event_dict[doc_id]["flags"] == "[DocumentFlagsAccessRemoved]" or \
+            event_dict[doc_id]["flags"] == "AccessRemoved", \
+            'Access Revoked flag is not tagged for document. Flag value: {}'.format(event_dict[doc_id]["flags"])
+
+    # Verifying if the docs, for which access has be revoked, are purged
+    doc_ids = db.getDocIds(cbl_db)
+    for sg_doc in docs_to_modify:
+        assert sg_doc["id"] not in doc_ids, "channel access removal didn't purge the docs from cbl db"
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.parametrize("num_of_docs", [
+    10,
+    100,
+])
+def test_replication_delete_event(params_from_base_test_setup, num_of_docs):
+    """
+    @summary:
```

Comment:
This summary looks same as above test. Can you change the summary according to the test?

Reply:
done